### PR TITLE
LT- 1114 goods types for open licence

### PR DESCRIPTION
**Pull request by 5pence**

### CODE REVIEW

File: applications/views.py [hunk 1]
```diff
@@ -73,33 +73,60 @@ def post(self, request):
...
-                site_on_application.save()
-
-            for external_location_on_draft in ExternalLocationOnDraft.objects.filter(draft=draft):
-                external_location_on_application = ExternalLocationOnApplication(
-                    external_location=external_location_on_draft.external_location,
-                    application=application)
-                external_location_on_application.save()
+            if application.licence_type == 'open_licence':
+                # Save associated end users, goods and sites
+                print('THERE')
```
Comment (cstobart) [inline applications/views.py]: remove

File: applications/views.py [hunk 2]
```diff
@@ -73,33 +73,60 @@ def post(self, request):
...
+                        application=application)
+                    site_on_application.save()
+
+                for external_location_on_draft in ExternalLocationOnDraft.objects.filter(draft=draft):
+                    external_location_on_application = ExternalLocationOnApplication(
+                        external_location=external_location_on_draft.external_location,
+                        application=application)
+                    external_location_on_application.save()
+
+            if application.licence_type == 'standard_licence':
```
Comment (cstobart) [inline applications/views.py]: might be safer to use an else if

File: applications/views.py [hunk 3]
```diff
@@ -73,33 +73,60 @@ def post(self, request):
...
-                    external_location=external_location_on_draft.external_location,
-                    application=application)
-                external_location_on_application.save()
+            if application.licence_type == 'open_licence':
+                # Save associated end users, goods and sites
+                print('THERE')
+                application.end_user = draft.end_user
+                application.save()
+
+                for goodstypes_on_draft in GoodOnDraft.objects.filter(draft=draft):
```
Comment (cstobart) [inline applications/views.py]: goods_types_on_draft instead of goods types_on_draft

File: conf/settings.py [hunk 4]
```diff
@@ -56,6 +56,8 @@
     'static.denial_reasons',
     'reversion',
     'drf_yasg',
+
```
Comment (cstobart) [inline conf/settings.py]: remove blank line

File: goodstype/migrations/0001_initial.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='GoodsType',
+            fields=[
+                ('id', models.UUIDField(default=uuid.uuid4, editable=False, primary_key=True, serialize=False)),
+                ('description', models.TextField(blank=True, default=None, max_length=280, null=True)),
+                ('is_good_controlled', models.BooleanField(blank=True, default=None, null=True)),
+                ('control_code', models.TextField(blank=True, default=None, null=True)),
```
Comment (cstobart) [inline goodstype/migrations/0001_initial.py]: This could be several control codes?

File: drafts/views/draft_goods.py [hunk 6]
```diff
@@ -6,11 +6,24 @@
...
+from goodstype.serializers import GoodsTypeSerializer
 from organisations.libraries.get_organisation import get_organisation_by_user
 
 
+class DraftGoodsType(APIView):
+    authentication_classes = (PkAuthentication,)
+    """
+    View goods belonging to a draft, or add one
+    """
+    def get(self, request, pk):
```
Comment (lukesa) [inline drafts/views/draft_goods.py]: add documentation
https://uktrade.atlassian.net/wiki/spaces/ILT/pages/746684421/Guidelines+for+LITE+API

File: drafts/views/draft_goods.py [hunk 7]
```diff
@@ -6,11 +6,24 @@
...
+from drafts.models import GoodOnDraft, Draft
 from drafts.serializers import GoodOnDraftBaseSerializer, GoodOnDraftViewSerializer
+from goodstype.models import GoodsType
+from goodstype.serializers import GoodsTypeSerializer
 from organisations.libraries.get_organisation import get_organisation_by_user
 
 
+class DraftGoodsType(APIView):
+    authentication_classes = (PkAuthentication,)
+    """
```
Comment (lukesa) [inline drafts/views/draft_goods.py]: docstring should be at 1st line of the class (before authentication_classes)

File: applications/views.py (hunk 2 above)
Comment (lukesa) [reply applications/views.py]: or switch statement instead of number of IF (with default action at the end)

File: goodstype/apps.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+from django.apps import AppConfig
```
Comment (lukesa) [inline goodstype/apps.py]: rename app to ```goods_type```

File: applications/views.py (hunk 2 above)
Comment (5pence) [reply applications/views.py]: Python doesn't have a switch statement @lukesa

File: goodstype/migrations/0001_initial.py (hunk 5 above)
Comment (5pence) [reply goodstype/migrations/0001_initial.py]: it's one per goodstype

File: goodstype/apps.py (hunk 8 above)
Comment (5pence) [reply goodstype/apps.py]: no switch statements exist in Python

File: applications/views.py (hunk 3 above)
Comment (antonhughes) [reply applications/views.py]: done

File: conf/settings.py (hunk 4 above)
Comment (antonhughes) [reply conf/settings.py]: done

File: goodstype/views.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                                status=status.HTTP_201_CREATED)
+
+        return JsonResponse(data={'errors': serializer.errors},
+                            status=status.HTTP_400_BAD_REQUEST)
+
+
+class GoodsTypeDetail(APIView):
+    authentication_classes = (PkAuthentication,)
+
+    def get(self, request, pk):
```
[the remark this answers is not in the document]
Comment (antonhughes) [reply goodstype/views.py]: done

File: goodstype/views.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+    authentication_classes = (PkAuthentication,)
+
+    def get(self, request):
+        description = request.GET.get('description', '')
+        goods = GoodsType.objects.filter()
+        serializer = GoodsTypeSerializer(goods, many=True)
+        return JsonResponse(data={'goods': serializer.data},
+                            safe=False)
+
+    def post(self, request):
```
[the remark this answers is not in the document]
Comment (antonhughes) [reply goodstype/views.py]: done

File: drafts/views/draft_goods.py (hunk 6 above)
Comment (antonhughes) [reply drafts/views/draft_goods.py]: done

File: goodstype/apps.py (hunk 8 above)
Comment (antonhughes) [reply goodstype/apps.py]: done

**Pull request merged**